crates/ironrdp-web/src/session.rs

```diff
@@ -216,13 +205,28 @@ impl SessionBuilder {
         self.clone()
     }
 
-    /// Optional
-    pub fn use_display_control(&self) -> SessionBuilder {
-        self.0.borrow_mut().use_display_control = true;
+    pub fn extension(&self, value: JsValue) -> SessionBuilder {
+        match serde_wasm_bindgen::from_value::<Extension>(value) {
+            Ok(value) => match value {
+                Extension::KdcProxyUrl(kdc_proxy_url) => self.0.borrow_mut().kdc_proxy_url = Some(kdc_proxy_url),
+                Extension::Pcb(pcb) => self.0.borrow_mut().pcb = Some(pcb),
+                Extension::DisplayControl(use_display_control) => {
+                    self.0.borrow_mut().use_display_control = use_display_control
+                }
+            },
+            Err(err) => error!("Provided JsValue is not a valid extension value: {err:?}"),
+        }
+
         self.clone()
     }
```

Comment:
thought: I’m not sure this is exactly what we want. At least it’s defined differently in the design document. But it does not look bad either. I need to think about it.

Reply:
Indeed. It's not the same you shared in the design document:
```rust
struct Extension<'ident> {
    ident: &'ident str,
    value: JsValue,
}
```
I think that instead we can encapsulate the identifier in a type. Rust and TS allow us to do this and this approach is more reliable, since we do not need to do manual matching of `ident` to the provided `JsValue`, and therefore this approach is less error-prone

Reply:
I like that!